docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,612 @@
+---
+
+layout: docs
+
+title:  "Connectable Operators"
+
+section: "chisel3"
+
+---
+
+## Terminology
+
+- "Chisel type" - a `Data` that is not bound to hardware, i.e. not a component. (more details [here](chisel-type-vs-scala-type)).
+  - E.g. `UInt(3.W)`, `new Bundle {..}`, `Vec(3, SInt(2.W))` are all Chisel types
+- `Aggregate` - a Chisel type or component that contains other Chisel types or components (i.e. `Vec`, `Record`, or `Bundle`)
+- `Element` - a Chisel type or component that does not contain other Chisel types or components (e.g. `UInt`, `SInt`, `Clock`, `Bool` etc.)
+- "component" - a `Data` that is bound to hardware (`IO`, `Reg`, `Wire`, etc.)
+  - E.g. `Wire(UInt(3.W))` is a component, whose Chisel type is `UInt(3.W)`
+- "member" - a Chisel type or component, or any of its children (could be an `Aggregate` or an `Element`)
+  - E.g. `Vec(3, UInt(2.W))(0)` is a member of the parent `Vec` Chisel type
+  - E.g. `Wire(Vec(3, UInt(2.W)))(0)` is a member of the parent `Wire` component
+  - E.g. `IO(Decoupled(Bool)).ready` is a member of the parent `IO` component
+- "relative alignment" - whether two members of the same component or Chisel type are aligned/flipped, relative to one another
+  - see section [below](#alignment-flipped-vs-aligned) for a detailed definition
+- "structural type check" - Chisel type `A` is structurally equivalent to Chisel type `B` if `A` and `B` have matching bundle field names and types (`Record` vs `Vector` vs `Element`), vector sizes, `Element` types (UInt/SInt/Bool/Clock etc))
+  - ignores relative alignment (flippedness)
+  - use `DataMirror.checkTypeEquivalence` to check this property
+- "alignment type check" - a Chisel type `A` matches alignment with another Chisel type `B` if every member of `A`'s relative alignment to `A` is the same as the structurally corresponding member of `B`'s relative alignment to `B`.
```

Comment:
sorry i am still confused. you define chisel type above not in terms of structural type or alignment type. is chisel type equivalence = structural type equivalalence + alignment type equivalence, or not?

Reply:
Yes, it is (+ width type equivalence). I need to update this, because there exists no way to actually just check these properties. Also.. its a bit hand-wave-y because technically you can drive `Reset` from `Bool`, but I'm not convinced they would (or should) structurally type check.

Reply:
I removed 'chisel type equivalence' because that actually isn't the thing that is checked.